packages/x-date-pickers/src/internals/hooks/useField/useFieldV6TextField.ts

```diff
@@ -162,6 +162,18 @@ export const useFieldV6TextField: UseFieldTextField<false> = (params) => {
               inputRef.current.setSelectionRange(selectionStart, selectionEnd);
             }
           }
+          setTimeout(() => {
+            // handle case when the selection is not updated correctly
+            // could happen on Android
+            if (
+              inputRef.current &&
+              inputRef.current === getActiveElement(document) &&
+              (inputRef.current.selectionStart !== selectionStart ||
+                inputRef.current.selectionEnd !== selectionEnd)
+            ) {
+              interactions.syncSelectionToDOM();
```

Comment:
I'm afraid of a possible endless recursion here. 🤔 
WDYT, do we need an additional safeguard against it here?

Reply:
Do you have a clearer idea of a scenario when this recursion might happen?
Tbh I don't really know how to add safeguards here :grimacing: